### PR TITLE
Adding better decal culling

### PR DESCRIPTION
**Pull request by mrieggeramzn**
Adding some lines to decal culling to improve accuracy on areas with high depth discontinuity

### BEFORE (1024 decals - Sponza LightCulling sample)
decals before
### AFTER
decals after

Signed-off-by: mrieggeramzn <mriegger@amazon.com>

### CODE REVIEW

**Comment by rgba16f** [comment]
Could the sphere radius sqr be estimated by `dot(decal.m_halfSize, decal.m_halfSize)`

**Review by rgba16f — APPROVED** [review]

**Comment by mrieggeramzn** [comment]
> Could the sphere radius sqr be estimated by `dot(decal.m_halfSize, decal.m_halfSize)`

Sounds good. Was playing around and it like it performs a bit better.

File: Gems/Atom/Feature/Common/Assets/Shaders/LightCulling/LightCulling.azsl [hunk 1]
```diff
@@ -368,16 +368,20 @@ void CullDecals(uint groupIndex, TileLightData tileLightData, float3 aabb_center
         float3 decalPosition = WorldToView_Point(decal.m_position); 
         
         // just wrapping a bounding sphere around a cube for now to get a minor perf boost. i.e. the sphere radius is sqrt(x*x + y*y + z*z)
-        // ATOM-4224 - try AABB-AABB and implement depth binning for the decals
+        // ATOM-4224 - try AABB-AABB 
         float maxHalfSize = max(max(decal.m_halfSize.x, decal.m_halfSize.y), decal.m_halfSize.z);
```
Comment (jeremyong-az) [inline Gems/Atom/Feature/Common/Assets/Shaders/LightCulling/LightCulling.azsl:372]: Note: `maxHalfSize` is unused now

Comment (mrieggeramzn) [reply Gems/Atom/Feature/Common/Assets/Shaders/LightCulling/LightCulling.azsl]: Thanks, good eye. Fixed now.

**Review by jeremyong-az — APPROVED** [review]

**Review by Adi-Amazon — APPROVED** [review]

**Pull request merged**